### PR TITLE
tiflash: update behavior under CBO mode

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add description about behavior under TiDB CBO mode.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2883
- Other reference link(s):

### CODE REVIEW

File: reference/tiflash/use-tiflash.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ SELECT * FROM information_schema.tiflash_replica WHERE TABLE_SCHEMA = '<db_name>
 
 In the result of above statement:
 
-* `AVAILABLE` indicates whether the TiFlash replicas of this table is available or not. `1` means available and `0` means unavailable.
+* `AVAILABLE` indicates whether the TiFlash replicas of this table is available or not. `1` means available and `0` means unavailable. Once the replicas become available, this status does not change. If you have used DDL statements to modify the number of replicas, the replication status is recalculated.
```
Comment (yikeke) [inline reference/tiflash/use-tiflash.md:75]:
```suggestion
* `AVAILABLE` indicates whether the TiFlash replicas of this table is available or not. `1` means available and `0` means unavailable. Once the replicas become available, this status does not change. If you use DDL statements to modify the number of replicas, the replication status will be recalculated.
```

File: reference/tiflash/use-tiflash.md [hunk 2]
```diff
@@ -118,6 +118,8 @@ explain analyze select count(*) from test.t;
 
 `cop[tiflash]` means that the task will be sent to TiFlash for processing. If you have not selected a TiFlash replica, you can try to update the statistics using the `analyze table` statement, and then check the result using the `explain analyze` statement.
 
+Note that if a table has only a single TiFlash replica and the node cannot provide service, queries in the smart mode will repeatedly retry. In this situation, you need to specify the engine or use the manual Hint to read data from TiKV.
```
Comment (yikeke) [inline reference/tiflash/use-tiflash.md:121]:
```suggestion
Note that if a table has only a single TiFlash replica and the related node cannot provide service, queries in the CBO mode will repeatedly retry. In this situation, you need to specify the engine or use the manual Hint to read data from TiKV.
```
See https://github.com/pingcap/docs-cn/pull/2883#pullrequestreview-400490430 for my suggestion.

**Comment by yikeke** [comment]
Please sign the CLA, or this PR cannot be merged. @solotzg

**Comment by solotzg** [comment]
@yikeke done

**Pull request merged**